### PR TITLE
Add unit tests for EventPublisherMediator and flow.statics.publishing

### PR DESCRIPTION
**Pull request by GDLMadushanka**
## Automation tests
 - Unit tests 
   > 73%, 66%

### CODE REVIEW

File: modules/core/src/test/java/org/apache/synapse/aspects/flow/statistics/publishing/PublishingPayloadTest.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    /**
+     * Test adding sample payload event
+     */
+    @Test
+    public void testAddEvent() {
+        PublishingPayloadEvent payloadEvent = new PublishingPayloadEvent();
+        payloadEvent.setAttribute(new Integer(1));
+        payloadEvent.setEventIndex(1);
+        Assert.assertTrue("Adding event should be successful",publishingPayload.addEvent(payloadEvent));
```
Comment (chanikag) [inline modules/core/src/test/java/org/apache/synapse/aspects/flow/statistics/publishing/PublishingPayloadTest.java]: Code formating issue: need a space after ,

File: modules/core/src/test/java/org/apache/synapse/mediators/eventing/EventPublisherMediatorTest.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        messageContext.setEnvironment(synapseEnvironment);
+        SynapseConfiguration synapseConfiguration = new SynapseConfiguration();
+        SynapseEventSource synapseEventSource = Mockito.mock(SynapseEventSource.class);
+        Mockito.doNothing().when(synapseEventSource).dispatchEvents(Mockito.any(MessageContext.class));
+        Map<String,SynapseEventSource> eventStoreMap = new HashMap<>();
+        eventStoreMap.put(EVENT_SOURCE_NAME,synapseEventSource);
+        synapseConfiguration.setEventSources(eventStoreMap);
+        messageContext.setConfiguration(synapseConfiguration);
+        eventPublisherMediator.setEventSourceName(EVENT_SOURCE_NAME);
+        Assert.assertTrue("mediation successful",eventPublisherMediator.mediate(messageContext));
```
Comment (bsenduran) [inline modules/core/src/test/java/org/apache/synapse/mediators/eventing/EventPublisherMediatorTest.java]: Same as mentioned above

File: modules/core/src/test/java/org/apache/synapse/mediators/eventing/EventPublisherMediatorTest.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    public void testMediate() {
+        final String EVENT_SOURCE_NAME = "testEventStore";
+        SynapseEnvironment synapseEnvironment = Mockito.mock(SynapseEnvironment.class);
+        MessageContext messageContext = new TestMessageContext();
+        messageContext.setEnvironment(synapseEnvironment);
+        SynapseConfiguration synapseConfiguration = new SynapseConfiguration();
+        SynapseEventSource synapseEventSource = Mockito.mock(SynapseEventSource.class);
+        Mockito.doNothing().when(synapseEventSource).dispatchEvents(Mockito.any(MessageContext.class));
+        Map<String,SynapseEventSource> eventStoreMap = new HashMap<>();
+        eventStoreMap.put(EVENT_SOURCE_NAME,synapseEventSource);
```
Comment (chanikag) [inline modules/core/src/test/java/org/apache/synapse/mediators/eventing/EventPublisherMediatorTest.java]: Code formating issue: need a space after ,

File: modules/core/src/test/java/org/apache/synapse/mediators/eventing/EventPublisherMediatorTest.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+public class EventPublisherMediatorTest {
+
+    private EventPublisherMediator eventPublisherMediator = new EventPublisherMediator();
+
+    /**
+     * Testing eventPublisherMediator with mocked event source
+     */
+    @Test
+    public void testMediate() {
+        final String EVENT_SOURCE_NAME = "testEventStore";
```
Comment (chanikag) [inline modules/core/src/test/java/org/apache/synapse/mediators/eventing/EventPublisherMediatorTest.java]: This is a method variable. So do not have to define it in block capitals.

File: modules/core/src/test/java/org/apache/synapse/aspects/flow/statistics/publishing/PublishingEventTest.java [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+        publishingEvent.setBeforePayload(BEFORE_PAYLOAD);
+        publishingEvent.setAfterPayload(AFTER_PAYLOAD);
+    }
+
+    /**
+     * Testing the overridden toSting method
+     */
+    @Test
+    public void testToString() {
+        String expected = "Component Type " + StatisticsConstants.FLOW_STATISTICS_ENDPOINT +
```
Comment (sdkottegoda) [inline modules/core/src/test/java/org/apache/synapse/aspects/flow/statistics/publishing/PublishingEventTest.java]: line wrapping should happen before an operator. Shall we move the '+' sign to the next line?

File: modules/core/src/test/java/org/apache/synapse/aspects/flow/statistics/publishing/PublishingEventTest.java [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Unit tests for PublishingEvent class
```
Comment (sdkottegoda) [inline modules/core/src/test/java/org/apache/synapse/aspects/flow/statistics/publishing/PublishingEventTest.java]: Doc comments need to end with a period. Shall we correct in all places?

**Pull request merged**